### PR TITLE
Support returning with sqlite3 and better-sqlite3

### LINKED ISSUE #4370: [SQLite] Support .returning in the latest SQLite
[`.returning()`](http://knexjs.org/#Builder-returning) doesn't work on SQLite, however it looks like their latest version does support it now: https://nalgeon.github.io/sqlite-3-35/

Any chance of getting this working in Knex? Thanks

### PR DESCRIPTION
**Pull request by simonplend**
The `RETURNING` SQL statement is now supported for insert and update queries.

Fixes #4370.

### CODE REVIEW

**Comment by simonplend** [comment]
@kibertoad Please can you approve the pending GitHub Actions workflows.

**Comment by simonplend** [comment]
Once this PR has been approved I'll update the documentation for `.returning` to reflect support for SQLite ([reference](https://github.com/knex/documentation/blob/main/src/guide/query-builder.md#returning)).

File: test/integration2/query/misc/additional.spec.js
Comment (kibertoad) [inline test/integration2/query/misc/additional.spec.js:279]: isSQLite should cover both isBetterSQlite and node-Sqlite3

File: lib/dialects/sqlite3/index.js
Comment (simonplend) [reply lib/dialects/sqlite3/index.js:127]: Yes, that's correct.

Comment (simonplend) [reply lib/dialects/sqlite3/index.js:127]: I've noticed that I now need to go and fix the the insert tests for sqlite.

**Comment by simonplend** [comment]
@kibertoad Sorry, just realised that there were a lot of tests I accidentally didn't run locally, so I missed a bunch of failures. Going to dig into them now.

**Comment by simonplend** [comment]
@kibertoad Think I've fixed all the tests now 🤞🏻

**Comment by simonplend** [comment]
@kibertoad All the tests are passing now, but Coveralls is reporting a -0.001% decrease in coverage. It looks like coverage fell for this file: https://coveralls.io/builds/51316627/source?filename=lib%2Fquery%2Fquerycompiler.js

I'm struggling to see how I can bring the coverage back up for that file — any ideas?

File: lib/dialects/sqlite3/index.js
Comment (simonplend) [reply lib/dialects/sqlite3/index.js:127]: I've altered this. Insert is no longer handled by the default case.

**Comment by kibertoad** [comment]
@simonplend Coverage is just a tool, there is no strict need to satisfy that check, although I'm curious as to what made that line no longer to be covered.

**Comment by simonplend** [comment]
@kibertoad Got it — I thought that's what was blocking it from being merged, but I think it was just that the PR is pending review approval.

> ...although I'm curious as to what made that line no longer to be covered.

I'm curious too, but I have no idea where to look!

If you're happy with the code in this PR I'll rebase the branch and tidy up the fixup commits.

**Comment by kibertoad** [comment]
yup, it looks great!

**Comment by simonplend** [comment]
@kibertoad Rebased and ready for review :)

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Thanks a lot!

**Comment by simonplend** [comment]
Thanks for the review! I'll open a corresponding documentation PR tomorrow.

**Pull request merged**
